rfc/0002/README.md

```diff
@@ -0,0 +1,169 @@
+    $PERSISTENT_DW_foo-persistent-storage=some-lustre
+```
+
+Peter submits the job to the WLM. WLM guides the job through the workflow states:
+1. Proposal: Rabbit software verifies the #DW directives. For the container directive, the requirements of Container Profile `foo`, that is `foo-local-storage` and `foo-persistent-storage`, are provided and correct.
+2. Setup: Since there is a jobdw `my-gfs2`, Rabbit software provisions this storage.
+3. Pre-Run:
+    1. Rabbit software generates a config map that corresponds to the storage requirements and runtime parameters.
+
+    ```yaml
+    kind: ConfigMap
+    apiVersion: v1
+    metadata:
+        name: my-job-container-my-foo
+    data:
+        $JOB_DW_foo-local-storage:             type=gfs2   mount-type=indexed-mount
+        $PERSISTENT_DW_foo-persistent-storage: type=lustre mount-type=mount-point
+    ```
+    2. Rabbit software duplicates the `foo` pod spec in the Container Profile and fills in the necessary volumes and config map.
+
+    ```yaml
+    kind: Pod
+    apiVersion: v1
+    metadata:
+        name: my-job-container-my-foo
+    spec:
+        containers:
+        # This section unchanged from Container Profile
+        - name: foo
+          image: foo:latest
+          command:
+            - /foo
+          volumeMounts:
+          - name: foo-local-storage
+            mountPath: /foo/local
+          - name: foo-persistent-storage
+            mountPath: /foo/persistent
+        
+          # This volumeMount added by Rabbit software
+          - name: nnf-config 
+            mountPath: /nnf/config
+
+        # volumes added by Rabbit software
+        volumes:
+        - name: foo-local-storage
+          hostPath:
+            path: /nnf/job/my-job/my-gfs2
+        - name: foo-persistent-storage
+          hostPath:
+            path: /nnf/persistent/some-lustre
+        - name: nnf-config
+          configMap:
+            name: my-job-container-my-foo
+    ```
+    3. Rabbit software starts the pods on Rabbit nodes
+
+
+
+Special Note: Indexed-Mount type
+--------------------------------
+
+When using a file system like GFS2, each compute is allocated its own Rabbit volume. The Rabbit software mounts a collection of mount paths with a common prefix and an ending indexed value. 
+
+Application authors must be aware that their desired mount-point really contains a collection of directories, one for each compute node. The mount point type can be known by consulting the config map values
+
+If we continue the example from above, the `foo` application would expect the foo-local-storage path of `/foo/local` to contain several directories
+
+```
+# ls /foo/local/*
+
+node-0
+node-1
+node-2
+...
+node-N
+```
+
+Node positions are ***not*** absolute locations. WLM could, in theory, select 6 physical compute nodes at physical location 1, 2, 3, 5, 8, 13, which would appear as directories /node-0 through /node-5 in the container.
```

Comment:
I think it is worth adding in an example showing how an NnfContainerProfile would reference the capacity tier lustre file system.  Basically the pod spec template is free form so they can just shove in the reference to the right PV/PVC.

Reply:
copy_in/copy_out just refer to it as a path in the directive parameters i.e `source=/lus/global/my-file.bin` There's no knowledge of PV/PVC anywhere and I would prefer to keep it that way. 

I'd rather the AUTHOR provide it in `spec.storages[]` and `spec.podSpec.containers[].volumeMounts[]` section in the profile and Rabbit software does the PV/PVC dance.